plainly-plugin/src/node/settings/settings.ts

```diff
@@ -0,0 +1,40 @@
+const fsPromises = require('fs/promises');
+const fs = require('fs');
+const path = require('path');
+
+import { settingsDirectory } from '../constants';
+import type { Settings } from '../types';
+
+export const defaultSettings: Settings = {};
+
+async function retrieveSettings() {
+  try {
+    const file = await fsPromises.readFile(
+      path.join(settingsDirectory, 'settings.json'),
```

Comment:
so twice `settings.json`, twice path created, wouldn't it be easier that settingsFileName and settingsPath are in the constants? and actually only path can be exported..